flask_errors/core.py

```
@@ -0,0 +1,28 @@
+"""
+    Contains the core functionality of Flask-Errors
+"""
+
+
+class HTTPError(Exception):
+
+    def __init__(self, message, identifier, status_code=None, payload=None):
+        Exception.__init__(self)
+        self.message = message
+        self.identifier = identifier
+        if status_code is not None:
+            self.status_code = status_code
+        self.payload = payload
+
+    def to_dict(self):
+        rv = dict(self.payload or ())
```

Consider a check of `self.payload` or `try-catch`.  
`payload` can be set to an arbitrary value by the caller, and this code may fail.
Any wrong value should result in a `ValueError` (or similar), not with a potentially confusing exception.

---

Considered an enhancement and moved to #9